### PR TITLE
feat: add SetMaxStackSize

### PR DESCRIPTION
**Pull request by rickyes**

### CODE REVIEW

**Comment by saghul** [comment]
Did you make changes so this is necessary? Note that this is the auxiliary tool to bundle JS files, the tjs binary already has an increased stack size.

**Comment by rickyes** [comment]
Hi，My idea is to change the size of the runtime stack with the command line argument

**Comment by saghul** [comment]
Yes, but you are changing it in qjsc.c, which is not the main interpreter. It's an auxiliary tool to create the bytecode that is then embedded in tjs. You may want to start looking at cli.c and vm.c.

**Comment by rickyes** [comment]
It has been modified.

File: src/cli.c [hunk 1]
```diff
@@ -248,6 +250,11 @@ int main(int argc, char **argv) {
                 exit_code = EXIT_INVALID_ARG;
                 goto exit;
             }
+            if (is_longopt(opt, "stack-size")) {
+                char *stack_size = get_option_value(arg, argc, argv, &optind);
+                runOption.stack_szie = (size_t) stack_size;
```
Comment (saghul) [inline src/cli.c]: you probably need `atoi` here, you can't just cast the string to a size_t

File: src/private.h [hunk 2]
```diff
@@ -63,6 +63,10 @@ struct TJSRuntime {
 #endif
 };
 
+typedef struct RunOption {
```
Comment (saghul) [inline src/private.h]: This shouldn't be private, let's move it to tjs.h

File: src/private.h [hunk 3]
```diff
@@ -63,6 +63,10 @@ struct TJSRuntime {
...
+    size_t stack_szie;
+} RunOption;
```
Comment (saghul) [inline src/private.h]: let's call this TJSRunOptions

File: src/tjs.h [hunk 4]
```diff
@@ -34,8 +34,9 @@
 
 typedef struct TJSRuntime TJSRuntime;
 
-TJSRuntime *TJS_NewRuntime(void);
+TJSRuntime *TJS_NewRuntime(RunOption *option);
 TJSRuntime *TJS_NewRuntime2(bool is_worker);
+TJSRuntime *TJS_NewRuntimeByOptions(bool is_worker, RunOption *option);
```
Comment (saghul) [inline src/tjs.h]: Please leave `TJSRuntime *TJS_NewRuntime(void);` alone, taking the default values.

Comment (saghul) [inline src/tjs.h]: Let's call this TJS_NewRuntime2 instead and skip the `is_worker` parameter. I'll make the current TJS_NewRuntime2 private. You can rename it to TJS_NewRuntimeWorker in the interim.

File: src/vm.c [hunk 5]
```diff
@@ -27,6 +27,8 @@
 
 #include <string.h>
 
+#define MAX_STACK_SIZE 1048576
```
Comment (saghul) [inline src/vm.c]: `TJS__DEFAULT_STACK_SIZE`

File: src/tjs.h (hunk 4 above)
Comment (rickyes) [reply src/tjs.h]: I plan to change it like this, what do you think:

``` c
TJSRuntime *TJS_NewRuntime(void);
TJSRuntime *TJS_NewRuntimeWorker(void);
TJSRuntime *TJS_NewRuntimeOption(bool is_worker, TJSRunOptions *option);
```

Comment (saghul) [reply src/tjs.h]: Sounds good!

**Comment by saghul** [comment]
Nice work, thank you very much!

**Comment by rickyes** [comment]
#117

**Pull request merged**